### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by Mahyar1982**
Let's check it

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
 });
 
 query5.on('error', function(error) {
-  // console.log(error);
+  // console.log(erro");
```
Comment (Mahyar1982) [inline app.js]: edit

File: app.js [hunk 2]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
 });
-
 query6.on('error', function(error) {
   // console.log(error);
 });
 app.get('/', function(req, res) {
   res.send('OK');
 });
 
+var bodyParser = require('body-parser');
```
Comment (Mahyar1982) [inline app.js]: check if can be removed

File: app.js [hunk 3]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
+app.use(bodyParser.json());       // to support JSON-encoded bodies
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+
+var admin = require("firebase-admin");
+var serviceAccount = require("./serviceAccountKey1.json");
```
Comment (Mahyar1982) [inline app.js]: remove 1

File: app.js [hunk 4]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount),
+  databaseURL: "https://free-time-c6774.firebaseio.com"
+  });
+
 app.post('/auth/', function(req, res) {
-  res.sendStatus(200);
-})
+  console.log("The coonection is OK");
```
Comment (Mahyar1982) [inline app.js]: edit

File: app.js [hunk 5]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
+admin.auth().verifyIdToken(req.body.token)
+  .then(function(decodedToken) {
+    var uid = decodedToken.uid;
+    console.log("The auth is ok");
```
Comment (Mahyar1982) [inline app.js]: res.send('OK');

File: app.js [hunk 6]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
+  }).catch(function(error) {
+    console.log("Error");
+    // Handle error
```
Comment (Mahyar1982) [inline app.js]: res.send(401);

File: app.js [hunk 7]
```diff
@@ -77,22 +77,42 @@ query4.on('error', function(error) {
...
+  });
+  res.end();
```
Comment (Mahyar1982) [inline app.js]: check if can be removed

**Review by m3hrad — APPROVED** [review]

**Pull request merged**